source/elements/oneDAL/source/glossary.rst

```diff
@@ -13,92 +13,113 @@ Machine learning terms
+
     Continuous feature
         A :term:`feature` with values in a domain of real numbers.
 
         Synonyms: quantitative feature, numerical feature.
 
+        Examples: a person's height, the price on the house.
+
     Dataset
         A collection of :term:`observations<Observation>`.
 
     Feature
         A particular property or quality of a real object or an event.
-        Has a defined type and domain.
+        Has a defined type and domain. In machine learning tasks, feature is
+        considered as input variable independent from others.
 
-        Synonyms: attribute, variable
-
-    Feature value
-        A single point in the domain of the :term:`feature`.
+        Synonyms: attribute, variable, input variable.
 
     Feature vector
         A vector that encodes information about single real object or an event.
-        Contains at least one :term:`feature value`.
+        Contains at least one :term:`feature`.
+
+        Examples: A rectangle can be described by two features: its width and height.
 
     Inference
-        A process of applying a :term:`trained<Training>` :term:`model` to the :term:`inference set` with a goal to
-        make outputs based on input :term:`feature vectors<Feature vector>`.
+        A process of applying a :term:`trained<Training>` :term:`model`
+        to the :term:`dataset` with a goal to make predictions of
+        :term:`responce` values based on input :term:`feature vectors<Feature vector>`.
 
         Synonyms: prediction.
 
     Inference set
-        An unlabeled :term:`dataset` used at the :term:`inference` stage to make an outputs from the :term:`model`.
+        A :term:`dataset` (usually without :term:`responses<Response>` present)
+        used at the :term:`inference` stage.
 
     Label
-        A :term:`feature` of some object or event which value is target to be predicted in :term:`supervised learning` task.
-        On the :term:`training` stage, the model which finds the best dependency between input :term:`feature vector` and
-        :term:`labels<Label>` is selected. :term:`Label` is one of outputs can be made by :term:`model`
-        on the :term:`inference` stage.
+        A :term:`response` with :term:`categorical<Categorical feature>` or :term:`ordinal<Ordinal feature>`
+        values. This is an output in :term:`classification` and :term:`clustering` tasks.
+
+        Examples: the spam-detection task has a binary label indicating whether the email is spam or not.
 
     Model
         A result of :term:`training` machine-learning algorithm that stores information necessary to run
         :term:`inference` on a new :term:`dataset`.
 
+        Examples: in linear regression algorithm, the model contains weight values for each input feature
+        and single bias value.
+
     Observation
-        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
```

Comment:
Just a feature vector and zero or more...

Reply:
Reformulated